### DISCUSSION TITLE
Xcode 13 package issues

### DISCUSSION THREAD

**Discussion by nmccann** [pointfreeco/isowords#120] [General]
Hi,

I wondering if you’ve had an opportunity to try the project with Xcode 13 Beta 3? This beta introduced a change that seems to be more strict about usage of APIs that aren’t valid in app extensions, but this rule is applied to all packages even if app extensions aren’t used. I’ve run into this with my own modular used project, as well as when running Isowords on the beta. There is at least one Swift forum thread discussing this [here](https://forums.swift.org/t/set-application-extension-api-only-on-a-spm-package/39333). The current workarounds are a bit ugly as they involve putting annotations throughout the project.

**Comment by mbrandonw** [comment]
Yes, we're seeing this issue too 😕. I was hoping adding the annotations wouldn't be so invasive, but it really explodes eventually.

I'll chime in on that thread to add another anecdote on the pain points.

**Reply by mbrandonw** [reply]
I got everything to build in this [diff](https://github.com/pointfreeco/isowords/compare/extenesion-annotation?expand=1). It's not _sooooo_ bad in our case, but it's still strange to litter these annotations throughout the code base.

**Reply by nmccann** [reply]
That diff isn’t as bad as the one in my project, so that may indicate that I need to reconsider how I have mine structured. But I’m definitely hopeful that Apple will provide some way to opt out of this, or to handle it more gracefully. I could see some wanting it to be very strict, but we should at least have the option.

Thanks for taking a look into the issue!
